[16.0][ADD] web_help

tarteo opened the pull request:
This module introduces a new way to guide the user through Odoo.
This module is created because **the standard Odoo tours:**

1. forces the user to create records;
2. cannot be reopened;
3. is only available for the admin user;
4. the bubbles are not obvious.

![demo](https://user-images.githubusercontent.com/10028499/220639790-9995d510-628e-4eef-a1d3-3c2aa8dac267.gif)


You can see the demo live by going to the users list view (Settings > Users & Companies > Users)

tarteo committed:
[ADD] web_help

[FIX] web_help: Run pre-commit

[FIX] web_help: Run pre-commit

[FIX] web_help: Run pre-commit

[FIX] web_help: Run pre-commit

[IMP] Easy changeable template and buttons texts, more demo, fixed readme.

[IMP] Easy changeable template and buttons texts, more demo, fixed readme.

[IMP] Easy changeable template and buttons texts, more demo, fixed readme.

[FIX] Grammar

simahawk commented:
@sbidoul do we have some problems w/ Runboat? I see is not appearing on my repos/PRs

simahawk commented:
> @sbidoul do we have some problems w/ Runboat? I see is not appearing on my repos/PRs

Seems to be only an issue w/ checks links on the PR since I can find the builds here https://runboat.odoo-community.org/webui/builds.html?repo=OCA/web&target_branch=16.0

simahawk commented:
@tarteo nice toy here :)

Questions:

1. I don't see anything popping up on runboat. Is there any manual action required?
2. how do you track if a help tour has already been used/discarded?

sbidoul commented:
@simahawk I don't see anything wrong with runboat here?

legalsylvain reviewed (commented):
Hi. 
I created a new user on runboat. test@test.com // test
I logged with the new user
I go to Settings > Users & Companies > Users
-> Nothing occures.

Did i missed something ? 

tested on chromium  110.0.5481.100 and Firefox 110.0.

legalsylvain commented on web_help/readme/USAGE.rst:
```diff
@@ -0,0 +1,6 @@
+  .. image:: ../static/description/demo.gif
+
+You can see the demo live by going to the users list view (Settings > Users & Companies > Users)
+It's easy to create your own guides, please refer to ``static/src/user_trip.js``
```
```suggestion
It's easy to create your own guides, please refer to ``static/src/user_trip.esm.js``
```

tarteo commented:
Hi @simahawk @legalsylvain!
I forgot to mention that you'll need to click the little '?' button next to the view switcher. I'll add this to the readme next week. 
The help tours are meant to be accessible even after the user already did them. But it would be nice addition to be able to make some tours activated automatically on the first time. Thanks!
Have a nice weekend!

legalsylvain commented:
Hi @tarteo.
> I forgot to mention that you'll need to click the little '?' button next to the view switcher

thanks for the explanation. works great ! 

I have however a design question. I don't think the place near the view switcher is optimal.

I think it should be better near the title of the page. In a tree view : 

![image](https://user-images.githubusercontent.com/3407482/221302857-44466868-295f-44a9-b757-d8ccd15fbac1.png)

**Reason :** Seeing all the different kind of view, the title seems to be the only place always displayed.

- in a classic form view the title (breadcrumb) is available, but the view switcher is not : 

![image](https://user-images.githubusercontent.com/3407482/221303120-baa92c36-3478-4d8c-84fb-2b316c2c75f4.png)

- In a pop up form view the title is available, but the view switcher is not : 

![image](https://user-images.githubusercontent.com/3407482/221303337-4ff21b36-68be-475b-a094-2abe9e1a57e9.png)

Any reason to have chosen the view switcher place ? technical ? UX ? 

Note : I agree that it is partly a matter of taste.

Regards.

legalsylvain commented on web_help/static/src/trip.xml:
```diff
@@ -0,0 +1,29 @@
+
+            <button
+                class="btn btn-lg btn-outline-light web_help_cb_button mr8"
+            >Got it</button>
```
```suggestion
            >Next</button>
```

Nitpicking + Not blocking : I find that "Got it" can mean "stop the trip", while "Next" is more explicit.

tarteo replied:
I've changed it so it's easy to change the button text or the whole template.

legalsylvain commented on web_help/static/src/trip.xml:
```diff
@@ -0,0 +1,29 @@
+            >Got it</button>
+            <button
+                class="btn btn-lg btn-outline-light web_help_close pull-right"
+            >Close</button>
```
```suggestion
            >Cancel</button>
```

Nitpicking + Not blocking : I find that "Close" can mean "close the current step", while "Cancel" is more explicit.

tarteo commented:
@legalsylvain I've made some changes:
* I've added the button also to the dialog
* Made it easy to change the template or the text of the buttons
* Fixed the readme
![changepassword](https://user-images.githubusercontent.com/10028499/221854545-04df765b-20ea-4447-abf6-1327b80eb4de.png)

legalsylvain commented:
Hi @tarteo. I tested again. Looks great !
For the button in the popup it is well displayed.
But for the Button in the tree view it is still near the sitcher. As a result it is not displayed in small screen (phone). Did you faced the same issue ?
I think putting the button near the title could fix this issue.

tarteo commented:
@legalsylvain For me it's visible also on small screens. I think it's web_responsive hiding I'll check it out and fix it.

tarteo committed:
[FIX] web_help: Make it work for small screens

[ADD] Roadmap

tarteo commented:
@legalsylvain It was not web_responsive that module is not yet ready for 16.0 :smile: It works also now for small screens.

![image](https://user-images.githubusercontent.com/10028499/222134242-3cf1f59c-10a9-4be1-9776-a90086eae657.png)

![image](https://user-images.githubusercontent.com/10028499/222134346-a9232a1c-0c9c-4915-95d7-8947c62e36ed.png)

simahawk commented:
> @simahawk I don't see anything wrong with runboat here?

![image](https://user-images.githubusercontent.com/347149/222170334-13a94cdc-b0ab-4a48-9e69-2485e774654a.png)

@sbidoul I mean this (here from a v14 build) :)

legalsylvain reviewed (approved):
Thanks for the changes.

functional review : All is OK for me. (no code review).

2 points : 
 
1. I still think that having the "?" button always in the same place (near the title) could be better for end user. (but not a blocking point, as it is also a matter of taste)
2. Do you though to have a shorcut key to display the next step. I just tested on runboat, and to display the next step, I have to click with the cursor on a button that is moving each time ! I think that catching "Enter" or "Space" input could be a great improvment, specially for long tour, with a lot of step. (you can add it in the roadmap.rst, if you think this point relevant).

thanks for your contribution and your reactivity during the review.

tarteo commented:
> 2\. Do you though to have a shorcut key to display the next step. I just tested on runboat, and to display the next step, I have to click with the cursor on a button that is moving each time ! I think that catching "Enter" or "Space" input could be a great improvment, specially for long tour, with a lot of step. (you can add it in the roadmap.rst, if you think this point relevant).

That sounds really nice. I've added it to the roadmap :)

tarteo commented:
/ocabot merge nobump
